aten/src/ATen/native/mps/OperationUtils.h

```diff
@@ -342,6 +342,6 @@ static inline void mtl_dispatch1DJob(id<MTLComputeCommandEncoder> encoder,
   [encoder dispatchThreads:size threadsPerThreadgroup:threadGroupSize];
 }
 
-id<MTLBuffer> generateKernelDataOffsets(id<MTLComputeCommandEncoder> commandEncoder, const TensorIteratorBase& iter);
+id<MTLBuffer> generateKernelDataOffsets(id<MTLComputeCommandEncoder> commandEncoder, const TensorIteratorBase& iter, bool use_64bit_index = false);
```

Comment:
Overall the change looks good to me . Just be careful not to enable 64 bit indexing unless absolutely needed as the performance of indexing arithmetic is low.

Reply:
Yes, that's the plan: 64-bit indexing is only enable if `iter.can_use_32bit_indexing()` is False (but let me add this information to the PR description)